tron/mesos.py

```diff
@@ -0,0 +1,272 @@
+import json
+import logging
+from urllib.parse import urlparse
+
+import requests
+from task_processing.runners.subscription import Subscription
+from task_processing.task_processor import TaskProcessor
+from twisted.internet.defer import logError
+
+from tron.actioncommand import ActionCommand
+from tron.utils.dicts import get_deep
+from tron.utils.queue import PyDeferredQueue
+
+TASK_LOG_FORMAT = '%(asctime)s %(name)s %(levelname)s %(message)s'
+# TODO: put in configs
+MESOS_MASTER_PORT = 5050
+DOCKERCFG_LOCATION = "file:///root/.dockercfg"
+MESOS_SECRET = ''
+MESOS_ROLE = '*'
+OFFER_TIMEOUT = 300
+
+log = logging.getLogger(__name__)
+_processor = None
+frameworks = {}  # TODO: improve
+
+
+def get_mesos_processor():
+    global _processor
+    if not _processor:
+        _processor = TaskProcessor()
+        _processor.load_plugin(provider_module='task_processing.plugins.mesos')
+    return _processor
+
+
+def get_mesos_leader(master_address):
+    url = "http://%s:%s/redirect" % (master_address, MESOS_MASTER_PORT)
+    response = requests.get(url)
+    return '{}:{}'.format(urlparse(response.url).hostname, MESOS_MASTER_PORT)
+
+
+def get_mesos_cluster(master_address):
+    global frameworks
```

Comment:
I'm not a huge fan of the global use here - should the dict of frameworks belong to the caller (and shutdown_frameworks take the dictionary as a parameter?)

Reply:
Hmm, the direct caller is an ActionRun. Keeping the dict there wouldn't enforce one framework per cluster, and it seems hard to get all the clusters back up to the MCP or trondaemon.

As a cleanup item, I was planning on making something similar to NodePoolRepository for frameworks to enforce the singleton-ness, instead of the current global. How does that sound?

Reply:
yep, that works.